tests/FactoryTest.php

```diff
@@ -335,4 +336,50 @@ public function it_lets_you_add_related_models_when_creating_multiple()
             $this->assertCount(5, $group->recipes);
         });
     }
+
+    /** @test */
+    public function it_lets_you_create_many_related_models_at_once()
+    {
+        Config::set('factories-reloaded.factories_namespace', 'ExampleAppTests\Factories');
+
+        $recipe = RecipeFactory::new()
+            ->with(Group::class, 'group')
+            ->with(Ingredient::class, 'ingredients', 3)
+            ->create();
+
+        $this->assertCount(1, Group::all());
+        $this->assertCount(3, Ingredient::all());
+        $this->assertTrue($recipe->group()->exists());
+        $this->assertEquals(3, $recipe->ingredients()->count());
+    }
+
+    /** @test */
+    public function it_lets_you_stack_relations()
+    {
+        Config::set('factories-reloaded.factories_namespace', 'ExampleAppTests\Factories');
+
+        $recipe = RecipeFactory::new()
+            ->with(Ingredient::class, 'ingredients')
+            ->with(Ingredient::class, 'ingredients', 3)
+            ->create();
+
+        $this->assertCount(4, Ingredient::all());
+        $this->assertEquals(4, $recipe->ingredients()->count());
+    }
+
+    /** @test */
+    public function it_lets_you_make_many_related_models_at_once()
+    {
+        Config::set('factories-reloaded.factories_namespace', 'ExampleAppTests\Factories');
+
+        $recipe = RecipeFactory::new()
+            ->with(Group::class, 'group')
+            ->with(Ingredient::class, 'ingredients', 3)
+            ->make();
+
+        $this->assertCount(1, Group::all());
+        $this->assertCount(0, Ingredient::all());
+        $this->assertTrue($recipe->group->exists);
```

Comment:
I think this "exists" property is only "true" when the model is persisted. It should work with changes in the BaseFactory and here checking if it is not null or if we get a group instance back.

Reply:
OK